pkg/userdata/flatcar/provider.go

```diff
@@ -73,37 +78,39 @@ func (p Provider) UserData(req plugin.UserDataRequest) (string, error) {
 		return "", fmt.Errorf("error extracting cacert: %v", err)
 	}
 
-	// We need to reconfigure rkt to allow insecure registries in case the hyperkube image comes from an insecure registry
-	var insecureHyperkubeImage bool
-	for _, registry := range req.InsecureRegistries {
-		if strings.Contains(req.HyperkubeImage, registry) {
-			insecureHyperkubeImage = true
-		}
-	}
-
 	if flatcarConfig.DisableAutoUpdate {
 		flatcarConfig.DisableLocksmithD = true
 		flatcarConfig.DisableUpdateEngine = true
 	}
 
+	kubeletImage := poseidonImage
+	lessThen118, err := semver.NewConstraint(lessThen118Check)
```

Comment:
no need to compile this each time and doing an error check as it is a constant, I would rather use a global variable initialised in an init block.

Reply:
I'm 100% do not agree with anything global or func init(). That's considered as bad coding practices by the community (for a good reason).

Reply:
@kron4eg Could you please elaborate a bit more about the "good reason" you mentioned referring to this specific case? Just to be clear, I was suggesting something like:

```go
var (
	lessThen118 = *constraintOrDie("< 1.18")
)

func constraintOrDie(c string) *semver.Constraints {
	o, err := semver.NewConstraint(c)
	if err != nil {
		panic(err)
	}
	return o
}
```

Compiling the constraint and checking the error every time would make more sense if it was dynamically provided, but as it is a constant it seems redundant to me.
The drawback I see with this approach is that `semver.Constraints` is mutable, so the theoretical risk of having it modified at runtime holds.

That said I'm not going to fight for this because it does not change much, it's just to understand your point.

Reply:
In general, go community condemn globals, I don't want to retype here all the reasons, you can find them here:
* https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html
* https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables


There is even a linter to check that your code has no globals https://github.com/leighmcculloch/gochecknoglobals

Reply:
Thx for the pointers, but IMO they don't add much to the discussion. I was more interested in your concerns regarding this specific case not the general one. Best practices can be useful tools, but it's important to separate fact from bias.  

> The drawback I see with this approach is that semver.Constraints is mutable, so the theoretical risk of having it modified at runtime holds.

This was the only theoretical drawback I found that could apply here, but after better looking it's not even valid, because the mutable internals of https://godoc.org/github.com/Masterminds/semver#Constraints are not exposed and no mutator is given.

If I did not convince you it's not a problem, I hope at least that your 100% become a 98% :-)